build/yaml/nightly-build-test-unix.yml

```diff
@@ -0,0 +1,120 @@
+    configuration: '$(BuildConfiguration)'
+
+- task: DotNetCoreCLI@2
+  displayName: 'Dotnet Publish TestBot'
+  inputs:
+    command: publish
+    publishWebProjects: false
+    projects: '$(System.DefaultWorkingDirectory)\tests\Microsoft.Bot.Builder.TestBot\Microsoft.Bot.Builder.TestBot.csproj'
+    arguments: '--output $(System.DefaultWorkingDirectory)\tests\Microsoft.Bot.Builder.TestBot\publishedbot'
+    zipAfterPublish: false
+    modifyOutputPath: false
+
+- task: AzureResourceGroupDeployment@2
+  displayName: 'Azure Deployment:Create Or Update Resource Group action on $(BotGroup)'
+  inputs:
+    azureSubscription: '$(AzureSubscription)'
+    resourceGroupName: '$(BotGroup)'
+    location: 'West US'
+    csmFile: FunctionalTests/ExportedTemplate/LinuxDotNet/template.json
+    overrideParameters: '-botName "$(BotName)" -appId "$(AppId)" -appSecret "$(AppSecret)"'
+
+- task: AzureCLI@1
+  displayName: 'Azure CLI '
+  inputs:
+    azureSubscription: '$(AzureSubscription)'
+    scriptLocation: inlineScript
+    inlineScript: 'call az bot directline create -n "$(BotName)" -g "$(BotGroup)" > "$(System.DefaultWorkingDirectory)\DirectLineCreate.json"'
+
+- script: |
+   Move $(System.DefaultWorkingDirectory)\tests\Microsoft.Bot.Builder.TestBot\DeploymentScripts\Linux\* $(System.DefaultWorkingDirectory)\tests\Microsoft.Bot.Builder.TestBot
+   git config --global user.name "BotBuilderDotNetPipeline"
+   git config --global user.email BotBuilderDotNet@Pipeline.com
+   git init
+   git add .
+   git commit -m "cibuildtest"
+   git remote add azure https://$(AzureDeploymentUser):$(AzureDeploymentPassword)@$(BotName).scm.azurewebsites.net:443/$(BotName).git
+   git push azure master
+  workingDirectory: '$(System.DefaultWorkingDirectory)\tests\Microsoft.Bot.Builder.TestBot'
+  displayName: 'Git Bot Deployment'
+
+- powershell: |
+   $json = Get-Content '$(System.DefaultWorkingDirectory)\DirectLineCreate.json' | Out-String | ConvertFrom-Json
+   $key = $json.properties.properties.sites.key
+   echo "##vso[task.setvariable variable=DIRECTLINE;]$key"
+   echo "##vso[task.setvariable variable=BOTID;]$(BotName)"
+  displayName: 'PowerShell Script'
+
+- powershell: |
+   echo '##vso[task.setvariable variable=TESTAPPID]$(AppId)'
+   echo '##vso[task.setvariable variable=TESTPASSWORD]$(AppSecret)'
+   echo '##vso[task.setvariable variable=LUISAPPID]$(LUISAPPIDSECRET)'
+   echo '##vso[task.setvariable variable=LUISSUBSCRIPTIONKEY]$(LUISSUBSCRIPTIONKEYSECRET)'
+  displayName: 'Set Environment Variables'
+
+- task: DotNetCoreCLI@2
+  displayName: 'dotnet test'
+  inputs:
+    command: test
+    projects: |
+     **/**Tests.csproj
+     !**/Microsoft.Bot.Builder.Classic.Tests.csproj
+     !**/Microsoft.Bot.ApplicationInsights.WebApi.Tests.csproj
+     !**/Microsoft.Bot.Builder.Integration.AspNet.WebApi.Tests.csproj
+    arguments: '-v n --configuration $(BuildConfiguration) --no-build --no-restore --filter TestCategory!=IgnoreInAutomatedBuild'
```

Comment:
You may want to change filter to '--filter "TestCategory!=IgnoreInAutomatedBuild&TestCategory!=FunctionalTests"'. See my comment in #3169.